### PR TITLE
Add Tom's regional DA dycore fix

### PR DESCRIPTION
**Pull request by junwang-noaa**
1) delz halo fix
2) changes for putting GSI anl into boundary file in a more robust manner in SAR DA system

Related PRs:
[fv3atm #132](https://github.com/NOAA-EMC/fv3atm/pull/132)
[ufs-weather-model #154](https://github.com/ufs-community/ufs-weather-model/pull/154)

### CODE REVIEW

**Review by EricRogers-NOAA — APPROVED** [review]

**Review by bensonr — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by XiaqiongZhou-NOAA — APPROVED** [review]

File: model/fv_regional_bc.F90 [hunk 1]
```diff
@@ -733,6 +735,24 @@ subroutine setup_regional_BC(Atm, dt_atmos                        &
         enddo
         enddo
       endif
+!
```
Comment (climbfuji) [inline model/fv_regional_bc.F90:738]: Is this specifically tailored to GFDL-MP? How about other microphysics schemes that use different tracers?

Comment (EricRogers-NOAA) [reply model/fv_regional_bc.F90:738]: No, it has been run for Thompson MP as well as GFDL MP.

Comment (climbfuji) [reply model/fv_regional_bc.F90:738]: Ok, thanks.

**Comment by danholdaway** [comment]
Thanks for including my changes Jun. Not sure where this happened and no problem if it doesn't make sense to include in this PR since this file wasn't touched and I can make another one but theres an additional change for when not using the INTERNAL_NML_FILE definition. In the file model/multi_gases.F90 at line 42 it should be `use fms_mod,           only: check_nml_error, open_namelist_file`. We'll work to get away from using this compiler definition since it's not recommended/supported but we could do with keeping it functional for now. I'll approve anyway in case you want to leave it out of this PR.

**Review by danholdaway — APPROVED** [review]

**Comment by danholdaway** [comment]
> Thanks for including my changes Jun. Not sure where this happened and no problem if it doesn't make sense to include in this PR since this file wasn't touched and I can make another one but theres an additional change for when not using the INTERNAL_NML_FILE definition. In the file model/multi_gases.F90 at line 42 it should be `use fms_mod, only: check_nml_error, open_namelist_file`. We'll work to get away from using this compiler definition since it's not recommended/supported but we could do with keeping it functional for now. I'll approve anyway in case you want to leave it out of this PR.

Apologies, copy paste error. It should be:
`use fms_mod, only: check_nml_error, open_namelist_file, close_file`
on that line

**Comment by junwang-noaa** [comment]
I see. We've about to commit for Tom's fix. I'd suggest to create another
dycore PR, we will get it in in next commit.

**Pull request merged**